### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by jtnord**
miscellaneous clean up

* use maven logging where it makes sense for debug output
* convert most resource management to `try-with-resources`
* remove unused imports
* as serialVersionUID to serialisable class (even if we never serialise it)
* remove the direct usage of guava

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jenkinsci/maven/plugins/hp/util/Utils.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class Utils {
+
+    /**
+     * Return an unmodifiable set whose contents are the unions of all the specified sets
+     * @param <T>
+     */
+    @SafeVarargs
+    public static <T> Set<T> unionOf(Set<T>... sets) {
+        HashSet<T> unionSet = new HashSet<>(sets[0]);
```
Comment (basil) [inline src/main/java/org/jenkinsci/maven/plugins/hp/util/Utils.java:15]: Effective Java Item 52: Refer to objects by their interfaces

```suggestion
        Set<T> unionSet = new HashSet<>(sets[0]);
```

File: src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java [hunk 2]
```diff
@@ -791,37 +792,19 @@ public Reader getReader(Reader fileReader, Properties filterProperties) {
...
-            // fix for MWAR-36, ensures that the parent dir are created first
-            Files.createDirectories(to.toPath().getParent());
-
-            if (encoding == null || encoding.length() < 1) {
-                fileReader = Files.newBufferedReader(from.toPath(), StandardCharsets.UTF_8);
-                fileWriter = Files.newBufferedWriter(to.toPath(), StandardCharsets.UTF_8);
-            } else {
-                FileInputStream instream = new FileInputStream(from);
-
-                FileOutputStream outstream = new FileOutputStream(to);
```
Comment (basil) [inline src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java:807]: There are some unused imports now that these have been deleted.

**Review by basil — APPROVED** [review]
Very nice!

**Comment by jtnord** [comment]
back to draft - something is completely broken in the resolution of what goes into a jpi

File: src/main/java/org/jenkinsci/maven/plugins/hp/util/Utils.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Set;
+
+public class Utils {
+
+    /**
+     * Return an unmodifiable set whose contents are the unions of all the specified sets
+     * @param <T>
+     */
+    @SafeVarargs
+    public static <T> Set<T> unionOf(Set<T>... sets) {
```
[the remark this answers is not in the document]
Comment (jtnord) [reply src/main/java/org/jenkinsci/maven/plugins/hp/util/Utils.java:14]: union - not intersection... 🤦

File: src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java (hunk 2 above)
Comment (jtnord) [reply src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java:807]: addressed in ce934abf061132ca12cb609e54cc7954591ccaf0

**Pull request merged**